functorch/_src/aot_autograd.py

```diff
@@ -1107,17 +1107,18 @@ def add_dupe_args(args):
 
     deduped_flat_args = remove_dupe_args(flat_args)
 
-    _fw_metadata, out, _num_aliasing_metadata_outs = run_functionalized_fw_and_collect_metadata(
-        lambda *args: flat_fn(*(add_dupe_args(args))),
-    )(*deduped_flat_args)
+    with enable_python_dispatcher():
```

Comment:
important bit #1: also enable python dispatcher during analysis when we run the forward (so the correct batch norm runs and we get accurate info on input mutations)